main.py

```diff
@@ -0,0 +1,146 @@
+        list_of_costumers_dicts = customers_data["customers"]
+        for customer in list_of_costumers_dicts:
+            if customer["customer_id"] == customer_id:
+                amount = int(input("Enter the amount you wish to withdraw"))
+                while amount > customer["balance"]:
+                    amount = int(input("Enter the amount you wish to withdraw"))
+                customer["balance"] -= amount
+                print(f"Your new balance is {customer['balance']}")
+                with open("customers.json", 'w') as customers_file_new:
+                    json.dump(customers_data, customers_file_new, indent=4)
+                return
+        else:
+            print("Invalid customer ID")
+            return withdrawal(login())
+
+
+def deposit(customer_id: int) -> None:
+    """
+    This function asks the user for the amount he wishes to deposit
+    param: customer_id: the customer ID of the user who wishes to deposit money
+    return: None
+    """
+    with open("customers.json", 'r') as customers_file:
+        customers_data = json.load(customers_file)
+        list_of_costumers_dicts = customers_data["customers"]
+        for customer in list_of_costumers_dicts:
+            if customer["customer_id"] == customer_id:
+                amount = int(input("Enter the amount you wish to deposit"))
+                customer["balance"] += amount
+                print(f"Your new balance is {customer['balance']}")
+                with open("customers.json", 'w') as customers_file_new:
+                    json.dump(customers_data, customers_file_new, indent=4)
+                return
+        else:
+            print("Invalid customer ID")
+            return deposit(login())
+
+
+def change_password(customer_id: int) -> None:
+    """
+    This function asks the user for his new password and changes it
+    param: customer_id: the customer ID of the user who wishes to change his password
+    return: None
+    """
+    with open("customers.json", 'r') as customers_file:
+        customers_data = json.load(customers_file)
+        list_of_costumers_dicts = customers_data["customers"]
+        for customer in list_of_costumers_dicts:
+            if customer["customer_id"] == customer_id:
+                password = int(input("Enter your new password"))
+                customer["password"] = password
+                print("Your password has been changed")
+                with open("customers.json", 'w') as customers_file_new:
+                    json.dump(customers_data, customers_file_new, indent=4)
+                return
+        else:
+            print("Invalid customer ID")
+            return change_password(login())
+
+
+def balance(customer_id: int) -> None:
+    """
+    This function prints the balance of the user
+    param: customer_id: the customer ID of the user who wishes to see his balance
+    return: None
+    """
+    with open("customers.json", 'r') as customers_file:

```

Comment:
Reading/writing from the customers file reoccur a couple of times. I think it should be extracted into functions and re-used